soc/nxp/imxrt/imxrt11xx/soc.c

```diff
@@ -456,6 +456,23 @@ static ALWAYS_INLINE void clock_init(void)
 	CLOCK_SetRootClock(kCLOCK_Root_Lpspi1, &rootCfg);
 #endif
 
+#ifdef CONFIG_VIDEO_MCUX_MIPI_CSI2RX
+	/* MIPI CSI-2 Rx connects to CSI via Video Mux */
+	CLOCK_EnableClock(kCLOCK_Video_Mux);
+	VIDEO_MUX->VID_MUX_CTRL.SET = VIDEO_MUX_VID_MUX_CTRL_CSI_SEL_MASK;
+
+	/* Configure MIPI CSI-2 Rx clocks */
+	rootCfg.div = 8;
+	rootCfg.mux = kCLOCK_CSI2_ClockRoot_MuxSysPll3Out;
+	CLOCK_SetRootClock(kCLOCK_Root_Csi2, &rootCfg);
+
+	rootCfg.mux = kCLOCK_CSI2_ESC_ClockRoot_MuxSysPll3Out;
+	CLOCK_SetRootClock(kCLOCK_Root_Csi2_Esc, &rootCfg);
+
+	rootCfg.mux = kCLOCK_CSI2_UI_ClockRoot_MuxSysPll3Out;
+	CLOCK_SetRootClock(kCLOCK_Root_Csi2_Ui, &rootCfg);
```

Comment:
This is probably where we should put the code writing to the PGMC register, if possible

Reply:
Sure- that is a good idea, and I'd support (at a minimum) making things like the video mux setting configurable in devicetree somehow. A simple video driver that reads the setting from the DT node could work well here. I just want to avoid SOC specific code in the driver for an IP block whenever possible, even if the IP is currently only present on one SOC we have enabled

Reply:
It's done [here](https://github.com/zephyrproject-rtos/zephyr/pull/71859).